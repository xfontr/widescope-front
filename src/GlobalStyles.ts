import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  body {
    margin: 0;
    min-height: 100vh;
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  button, input {
    font: inherit;
    border: none;
  }
`;

export default GlobalStyles;
