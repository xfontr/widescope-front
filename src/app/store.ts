import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import { userDataReducer } from "../store/slices/userData/userDataSlice";
import { userReducer } from "../store/slices/user/userSlice";

export const store = configureStore({
  reducer: {
    user: userReducer,
    userData: userDataReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
