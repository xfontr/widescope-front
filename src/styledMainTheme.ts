const styledMainTheme = {
  colors: {
    primary: "rgb(255, 255, 255)",
    primaryDarker: "rgb(238, 238, 238)",

    secondary: "rgb(25, 29, 35)",
    secondaryBrigther: "rgba(25, 29, 35, 0.7)",
  },

  shapes: {
    radiusSmall: "0.75rem",
    radiusBig: "1.5rem",

    shortShadow: "6px 6px 18px 6px rgba(0, 0, 0, 0.1)",
    longShadow: "10px 10px 30px 10px rgba(0, 0, 0, 0.1)",

    rectangularContainer: "0.8rem 1.5rem 0.8rem 1.5rem",
  },

  spacing: {
    gapSmall: "1rem",
    gapBig: "1.5rem",

    paddingSmall: "0.8rem",
    paddingBig: "1.5rem",
  },

  breakpoints: {
    small: "390px",
    big: "700pxpx",
    large: "1050px",
  },
};

export default styledMainTheme;
