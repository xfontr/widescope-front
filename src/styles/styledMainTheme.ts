const styledMainTheme = {
  colors: {
    primary: "rgb(255, 255, 255)",
    primaryDarker: "rgb(238, 238, 238)",

    secondary: "rgb(25, 29, 35)",
    secondaryBrigther: "rgba(25, 29, 35, 0.7)",

    error: "rgb(179, 120, 120)",
    success: "green",
  },

  shapes: {
    radiusSmall: "0.4rem",
    radiusBig: "1.5rem",

    shortShadow: "6px 6px 18px 6px rgba(0, 0, 0, 0.1)",
    longShadow: "10px 10px 30px 10px rgba(0, 0, 0, 0.1)",

    rectangularContainer: "0.7rem 1.6rem",
  },

  spacing: {
    gapSmall: "1.7rem",
    gapBig: "2.5rem",

    paddingSmall: "0.8rem",
    paddingBig: "1.8rem",
  },

  breakpoints: {
    small: "390px",
    big: "700px",
    large: "1250px",
  },
};

export default styledMainTheme;
