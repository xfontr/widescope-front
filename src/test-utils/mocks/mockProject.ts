import { IProject } from "../../types/project";

const mockProject: IProject = {
  id: "630e31ecb968115ba6a3e29f",
  name: "Project123",
  description: "The description",
  technologies: ["React", "Express"],
  repository: "www.google.com",
  author: "username",
  authorId: "id",
  logo: "ProjectLogo.png",
  logoBackup: "BackupProjectLogo.png",
  creationDate: "1662627402346",
};

export default mockProject;
