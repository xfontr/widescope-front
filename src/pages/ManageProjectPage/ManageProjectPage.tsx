import { useState } from "react";
import { useParams } from "react-router-dom";
import { useAppSelector } from "../../app/hooks";
import ProjectForm from "../../components/ProjectForm/ProjectForm";

interface ManageProjectPageProps {
  isCreate: boolean;
}

const ManageProjectPage = ({
  isCreate,
}: ManageProjectPageProps): JSX.Element => {
  const { projectId } = useParams();
  const requestedProject = useAppSelector((state) =>
    state.userData.projects.find(({ id }) => id === projectId)
  );

  const [project] = useState(requestedProject);

  return (
    <>
      <h2 className="page__title">
        {isCreate ? "New project" : "Update project"}
      </h2>

      {!isCreate && project && (
        <ProjectForm isCreate={isCreate} project={project} />
      )}
      {!isCreate && !project && <span>Loading the requested project...</span>}
      {isCreate && <ProjectForm isCreate={isCreate} />}
    </>
  );
};

export default ManageProjectPage;
