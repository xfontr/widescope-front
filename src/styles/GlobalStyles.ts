import { createGlobalStyle } from "styled-components";
import styledMainTheme from "./styledMainTheme";

const GlobalStyles = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  body {
    margin: 0;
    min-height: 100vh;
    background-color: ${styledMainTheme.colors.primaryDarker};
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  button, input {
    font: inherit;
    border: none;
  }
`;

export default GlobalStyles;
