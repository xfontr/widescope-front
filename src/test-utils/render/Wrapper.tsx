import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { store } from "../../app/store";
import GlobalStyles from "../../styles/GlobalStyles";
import styledMainTheme from "../../styles/styledMainTheme";

interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return (
    <ThemeProvider theme={styledMainTheme}>
      <Provider store={store}>
        <BrowserRouter>
          <GlobalStyles />
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>
  );
};

export default Wrapper;
