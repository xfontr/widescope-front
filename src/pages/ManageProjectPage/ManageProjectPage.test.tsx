import { render, screen } from "@testing-library/react";
import mockProject from "../../test-utils/mocks/mockProject";
import { WrapperWithMockStore } from "../../test-utils/render/Wrapper";
import ManageProjectPage from "./ManageProjectPage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import styledMainTheme from "../../styles/styledMainTheme";
import { store } from "../../app/store";
import { ThemeProvider } from "styled-components";
import { BrowserRouter } from "react-router-dom";
import GlobalStyles from "../../styles/GlobalStyles";

let mockProjectId = mockProject.id;

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({ projectId: mockProjectId }),
}));

describe("Given a ManageProjectPage component", () => {
  describe("When instantiated as a create project page", () => {
    test("Then it should match the snapshot", () => {
      const expectedPage = renderer.create(
        <Provider store={store}>
          <ThemeProvider theme={styledMainTheme}>
            <BrowserRouter>
              <GlobalStyles />
              <ManageProjectPage isCreate={true} />
            </BrowserRouter>
          </ThemeProvider>
        </Provider>
      );

      expect(expectedPage).toMatchSnapshot();
    });

    test("Then it should show a title and the create a project form", () => {
      render(<ManageProjectPage isCreate={true} />, {
        wrapper: WrapperWithMockStore,
      });

      const page = [
        screen.getByRole("heading", { name: "New project", level: 2 }),
        screen.getByRole("heading", {
          name: "Tell us about your project",
          level: 3,
        }),
      ];

      page.forEach((element) => expect(element).toBeInTheDocument());
    });
  });

  describe("When instantiated as a update project page", () => {
    test("Then it should match the snapshot", () => {
      const expectedPage = renderer.create(
        <Provider store={store}>
          <ThemeProvider theme={styledMainTheme}>
            <BrowserRouter>
              <GlobalStyles />
              <ManageProjectPage isCreate={false} />
            </BrowserRouter>
          </ThemeProvider>
        </Provider>
      );

      expect(expectedPage).toMatchSnapshot();
    });

    test("Then it should show a title and the create a project form", () => {
      render(<ManageProjectPage isCreate={false} />, {
        wrapper: WrapperWithMockStore,
      });

      const page = [
        screen.getByRole("heading", { name: "Update project", level: 2 }),
        screen.getByRole("heading", {
          name: `Update ${mockProject.name}`,
          level: 3,
        }),
      ];

      page.forEach((element) => expect(element).toBeInTheDocument());
    });

    test("Then it should show a loading message if the project is not found yet", () => {
      mockProjectId = "";

      render(<ManageProjectPage isCreate={false} />, {
        wrapper: WrapperWithMockStore,
      });

      const loadingMessage = screen.getByText(
        "Loading the requested project..."
      );

      expect(loadingMessage).toBeInTheDocument();
    });
  });
});
