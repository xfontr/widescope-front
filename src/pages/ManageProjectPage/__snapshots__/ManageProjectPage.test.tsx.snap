// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ManageProjectPage component When instantiated as a create project page Then it should match the snapshot 1`] = `
Array [
  <h2
    className="page__title"
  >
    New project
  </h2>,
  <form
    className="sc-bczRLJ jzjTXG"
    data-testid="form"
    onSubmit={[Function]}
  >
    <header
      className="sc-gsnTZi kjzfXG"
    >
      <h3
        className="form__title"
      >
        Tell us about your project
      </h3>
    </header>
    <div
      className="sc-dkzDqf bRizeo"
    >
      <label
        className="sc-hKMtZM chFrlm"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="sc-eCYdqJ dwpIGp"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Music app"
        type="text"
        value=""
      />
    </div>
    <div
      className="sc-dkzDqf bRizeo"
    >
      <label
        className="sc-hKMtZM chFrlm"
        htmlFor="repository"
      >
        Repository URL
      </label>
      <input
        className="sc-eCYdqJ dwpIGp"
        id="repository"
        name="repository"
        onChange={[Function]}
        placeholder="http://www.github.com/music-app"
        type="text"
        value=""
      />
    </div>
    <div
      className="sc-dkzDqf bRizeo area"
    >
      <label
        className="sc-hKMtZM chFrlm"
        htmlFor="logo"
      >
        Project logo
      </label>
      <input
        className="sc-eCYdqJ dwpIGp"
        id="logo"
        name="logo"
        onChange={[Function]}
        type="file"
        value=""
      />
    </div>
    <div
      className="sc-dkzDqf bRizeo"
    >
      <label
        className="sc-hKMtZM chFrlm"
        htmlFor="technologyFront"
      >
        Frontend main library or framework
      </label>
      <input
        className="sc-eCYdqJ dwpIGp"
        id="technologyFront"
        name="technologyFront"
        onChange={[Function]}
        placeholder="React"
        type="text"
        value=""
      />
    </div>
    <div
      className="sc-dkzDqf bRizeo"
    >
      <label
        className="sc-hKMtZM chFrlm"
        htmlFor="technologyBack"
      >
        Backend main library or framework
      </label>
      <input
        className="sc-eCYdqJ dwpIGp"
        id="technologyBack"
        name="technologyBack"
        onChange={[Function]}
        placeholder="Express"
        type="text"
        value=""
      />
    </div>
    <div
      className="sc-dkzDqf bRizeo area"
    >
      <label
        className="sc-hKMtZM chFrlm"
        htmlFor="description"
      >
        Description
      </label>
      <textarea
        className="sc-eCYdqJ dwpIGp"
        id="description"
        name="description"
        onChange={[Function]}
        placeholder="Music app is about sharing music between users"
        type=""
        value=""
      />
    </div>
    <div
      className="sc-iBkjds hTcyAC"
    >
      <ul
        className="error-list"
      />
    </div>
    <footer
      className="sc-jSMfEi etMdQu"
    >
      <button
        className="sc-gKXOVf fgJUzp undefined button--default "
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
        type="submit"
      >
        Create project
      </button>
    </footer>
  </form>,
]
`;

exports[`Given a ManageProjectPage component When instantiated as a update project page Then it should match the snapshot 1`] = `
Array [
  <h2
    className="page__title"
  >
    Update project
  </h2>,
  <span>
    Loading the requested project...
  </span>,
]
`;
